### PR TITLE
Refactor/remove aql cxx tests

### PR DESCRIPTION
**Pull request by Simran-B**
### Scope & Purpose

Remove useless JS tests that explicitly test C++ implementations of AQL functions. All built-in functions except V8() have a C++ implementation only nowadays, so there is no need to test the functions with and without NOOPT().

I'm unsure whether the arithmetic tests should test the behavior with and without NOOPT(). There's also one NOOPT() use that I left in because I don't know if it's still relevant: https://github.com/arangodb/arangodb/blob/e2065fedf2cd99d26c47d8544943365a991925b6/tests/js/server/aql/aql-functions.js#L4333-L4342

- [x] :hammer: Refactoring/simplification

### Checklist

- No user-visible changes
- Code coverage is presumably unchanged
- No backports

### CODE REVIEW

**Comment by jsteemann** [comment]
@Simran-B : for the arithmetic tests, there is actually a difference between using NOOPT(...) and ...
There should not be a difference for calls to AQL functions, but the arithmetic operators are different.

**Comment by Simran-B** [comment]
Undid the changes to the arithmetic tests.

File: tests/js/server/aql/aql-functions.js [hunk 1]
```diff
@@ -1406,126 +1085,50 @@ function ahuacatlFunctionsTestSuite () {
...
-      var expected = [ { "abc" : [ 1, 2, 3 ], "chicken" : "is a language", "empty" : false, "foo" : "bar", "foobar" : "baz", "quarter" : 1, "quux" : 123, "value" : true, "year" : 2010 } ];
-      var actual = getQueryResults("RETURN NOOPT(MERGE([ { quarter: 1, year: 2010 }, { foo: 'bar' }, { chicken: 'is a language' }, { }, { foobar: 'baz' }, { quux: 123, value: true }, { empty: false }, { }, { abc: [1,2,3] } ]))");
-      assertEqual(expected, actual);
-    },
-
-////////////////////////////////////////////////////////////////////////////////
-/// @brief test merge function
-////////////////////////////////////////////////////////////////////////////////
-
-    testMergeInvalid : function () {
```
Comment (jsteemann) [inline tests/js/server/aql/aql-functions.js:1470]: Shall we keep these?

File: tests/js/server/aql/aql-functions.js [hunk 2]
```diff
@@ -1406,126 +1085,50 @@ function ahuacatlFunctionsTestSuite () {
...
-/// @brief test merge_recursive function
-////////////////////////////////////////////////////////////////////////////////
-    
-    testMergeRecursive1 : function () {
-      var doc1 = "{ \"black\" : { \"enabled\" : true, \"visible\": false }, \"white\" : { \"enabled\" : true}, \"list\" : [ 1, 2, 3, 4, 5 ] }";
-      var doc2 = "{ \"black\" : { \"enabled\" : false }, \"list\": [ 6, 7, 8, 9, 0 ] }";
-
-      var expected = [ { "black" : { "enabled" : false, "visible" : false }, "list" : [ 6, 7, 8, 9, 0 ], "white" : { "enabled" : true } } ];
-
-      var actual = getQueryResults("RETURN MERGE_RECURSIVE(" + doc1 + ", " + doc2 + ")");
```
Comment (jsteemann) [inline tests/js/server/aql/aql-functions.js:1520]: Shall we keep this particular test?

File: tests/js/server/aql/aql-functions.js [hunk 3]
```diff
@@ -2905,375 +1960,124 @@ function ahuacatlFunctionsTestSuite () {
...
-        actual = getQueryResults(q, { value: input });
-        assertEqual([ [ 1, 2, 3, 4, 5, 6, 7, 8, 9 ] ], actual);
-      }
-    },
-
-////////////////////////////////////////////////////////////////////////////////
-/// @brief test min function
-////////////////////////////////////////////////////////////////////////////////
-    
-    testMin1 : function () {
```
Comment (jsteemann) [inline tests/js/server/aql/aql-functions.js:2958]: Is this test and some of the following tests just deleted? They don't seem to use NOOPT, and I don't replacements for them.
But I may have simply overlooked it given the size of the PR.

File: tests/js/server/aql/aql-functions.js (hunk 2 above)
Comment (Simran-B) [reply tests/js/server/aql/aql-functions.js:1520]: `testMergeRecursive1` tests `MERGE_RECURSIVE(doc1, doc2)`, 
`testMergeRecursive2` tests `MERGE_RECURSIVE(doc2, doc1`).
Seems meaningful to me? Or do you refer to the naming of the attribute keys (black/white)?

File: tests/js/server/aql/aql-functions-string.js [hunk 4]
```diff
@@ -1812,8 +1598,8 @@ function ahuacatlStringFunctionsTestSuite () {
 // //////////////////////////////////////////////////////////////////////////////
 
   testConcatList2: function () {
-    var expected = [ '["the","Quick","",null,"Brown",null,"アボカド","Jumps","名称について"]' ];
-    var actual = getQueryResults(`FOR r IN [ 1 ] return CONCAT([ 'the', 'Quick', '', null, 'Brown', null, 'アボカド', 'Jumps', '名称について' ], null)`);
+    var expected = [ '["the","Quick","",null,"Brown",null,"アボカド","Jumps","名称について"]false' ];
```
Comment (jsteemann) [inline tests/js/server/aql/aql-functions-string.js:1601]: Why the `false` here?
```suggestion
    var expected = [ '["the","Quick","",null,"Brown",null,"アボカド","Jumps","名称について"]' ];
```

File: tests/js/server/aql/aql-functions.js (hunk 3 above)
Comment (Simran-B) [reply tests/js/server/aql/aql-functions.js:2958]: It still exists: https://github.com/arangodb/arangodb/blob/935964279127ef0a6db6b2063736e4fdb1ef4afb/tests/js/server/aql/aql-functions.js#L1992-L1996

Comment (jsteemann) [reply tests/js/server/aql/aql-functions.js:2958]: Ok.

File: tests/js/server/aql/aql-functions.js (hunk 2 above)
Comment (jsteemann) [reply tests/js/server/aql/aql-functions.js:1520]: No, all good. The changes were simply too big to review sanely.

**Pull request merged**